# Исследование надежности заемщиков.
**Описание проекта**

Заказчик — кредитный отдел банка. Банку нужно определить как влияет семейное положение клиента на погашение кредита.
    
**Цель исследования** — проверьте две гипотезы:
1. Влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок.

**Ход исследования**

Входные данные от банка — статистика о платёжеспособности клиентов.

Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.
 
Таким образом, исследование пройдёт в три этапа:
 1. Обзор данных.
 2. Предобработка данных.
 3. Проверка гипотез.



## Обзор данных

In [1]:
# импорт библиотеки pandas
import pandas as pd

C:\Users\yerke\Anaconda3\lib\site-packages\pandas\compat\_optional.py:138: UserWarning: Pandas requires version '2.7.0' or newer of 'numexpr' (version '2.6.9' currently installed).
  warnings.warn(msg, UserWarning)


In [2]:
# чтение файла с данными и сохранение в df
df = pd.read_csv('data/data.csv')

In [3]:
# получение первых 10 строк таблицы df
df.head(10)

,children,days_employed,dob_years,education,education_id,family_status,family_status_id,gender,income_type,debt,total_income,purpose
0,1,-8437.673028,42,высшее,0,женат / замужем,0,F,сотрудник,0,253875.639453,покупка жилья
1,1,-4024.803754,36,среднее,1,женат / замужем,0,F,сотрудник,0,112080.014102,приобретение автомобиля
2,0,-5623.422610,33,Среднее,1,женат / замужем,0,M,сотрудник,0,145885.952297,покупка жилья
3,3,-4124.747207,32,среднее,1,женат / замужем,0,M,сотрудник,0,267628.550329,дополнительное образование
4,0,340266.072047,53,среднее,1,гражданский брак,1,F,пенсионер,0,158616.077870,сыграть свадьбу
5,0,-926.185831,27,высшее,0,гражданский брак,1,M,компаньон,0,255763.565419,покупка жилья
6,0,-2879.202052,43,высшее,0,женат / замужем,0,F,компаньон,0,240525.971920,операции с жильем
7,0,-152.779569,50,СРЕДНЕЕ,1,женат / замужем,0,M,сотрудник,0,135823.934197,образование
8,2,-6929.865299,35,ВЫСШЕЕ,0,гражданский брак,1,F,сотрудник,0,95856.832424,на проведение свадьбы
9,0,-2188.756445,41,среднее,1,женат / замужем,0,M,сотрудник,0,144425.938277,покупка жилья для семьи


In [4]:
# получение общей информации о данных в таблице df
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


Итак, в таблице 12 столбцов. Разные типы данных имеются в столбцах — `object` (5 столбцов), `int` (5 столбцов), `float` (2 столбца).

Согласно документации к данным:

* `children` — количество детей в семье
* `days_employed` — общий трудовой стаж в днях
* `dob_years` — возраст клиента в годах
* `education` — уровень образования клиента
* `education_id` — идентификатор уровня образования
* `family_status` — семейное положение
* `family_status_id` — идентификатор семейного положения
* `gender` — пол клиента
* `income_type` — тип занятости
* `debt` — имел ли задолженность по возврату кредитов
* `total_income` — ежемесячный доход
* `purpose` — цель получения кредита

**Выводы**

В каждой строке таблицы — данные о клиенте. Например, количество детей в семье, семейное положение, пол клиента. Также имеются информация о возрасте клиента в годах,  уровне образования клиента и тип занятости.   

Предварительно можно утверждать, что, данных достаточно для проверки гипотез. Но встречаются пропуски в данных.

In [5]:
df.describe()

,children,days_employed,dob_years,education_id,family_status_id,debt,total_income
count,21525.000000,19351.000000,21525.000000,21525.000000,21525.000000,21525.000000,1.935100e+04
mean,0.538908,63046.497661,43.293380,0.817236,0.972544,0.080883,1.674223e+05
std,1.381587,140827.311974,12.574584,0.548138,1.420324,0.272661,1.029716e+05
min,-1.000000,-18388.949901,0.000000,0.000000,0.000000,0.000000,2.066726e+04
25%,0.000000,-2747.423625,33.000000,1.000000,0.000000,0.000000,1.030532e+05
50%,0.000000,-1203.369529,42.000000,1.000000,0.000000,0.000000,1.450179e+05
75%,1.000000,-291.095954,53.000000,1.000000,1.000000,0.000000,2.034351e+05
max,20.000000,401755.400475,75.000000,4.000000,4.000000,1.000000,2.265604e+06


## Предобработка данных

In [6]:
# подсчёт доли пропусков
df.isna().mean()

children            0.000000
days_employed       0.100999
dob_years           0.000000
education           0.000000
education_id        0.000000
family_status       0.000000
family_status_id    0.000000
gender              0.000000
income_type         0.000000
debt                0.000000
total_income        0.100999
purpose             0.000000
dtype: float64

В основном `NaN` значение обнаружено в столбцах с пропущенными значениями. В столбцах `days_employed` и `total_income` доля пропусков составляет 10%. Также, пропуски для этих столбцов является одинаковыми, то есть если значение в столбце `days_employed` является пустым то соответственно значение в столбце `total_income` является тоже пустым, и наоборот. Возможные причины появления пропусков в данных могут быть:
* клиент является пенсионером то есть не имеет дохода только пенсия
* клиент не имеет какое то точное число заработка
* клиент может быть безработным

In [7]:
# Замена пропусков с медианным значением по столбцу days_employed
days_employed_median = df['days_employed'].median()
df['days_employed'].fillna(days_employed_median, inplace=True)

In [8]:
# Замена пропусков с медианным значением по столбцу total_income
total_income_median = df['total_income'].median()
df['total_income'].fillna(total_income_median, inplace=True)

Замена пропусков с медианным значением по столбцу является очень распрастраненным методом для количественных переменных.

In [9]:
# проверочный подсчёт доли пропусков
df.isna().mean()

children            0.0
days_employed       0.0
dob_years           0.0
education           0.0
education_id        0.0
family_status       0.0
family_status_id    0.0
gender              0.0
income_type         0.0
debt                0.0
total_income        0.0
purpose             0.0
dtype: float64

### Проверка данных на аномалии и исправления.

В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. Например, отрицательное количество дней трудового стажа в столбце days_employed. Для реальных данных это нормально. 

В столбце `children` имеются отрицательное значение -1, и можно их заменить средним значением. Также имеются клиенты которые имеет 20 детей что является аномалии, следовательно можно их заменить средним значением.

In [10]:
df[df['children'] < 0]

,children,days_employed,dob_years,education,education_id,family_status,family_status_id,gender,income_type,debt,total_income,purpose
291,-1,-4417.703588,46,среднее,1,гражданский брак,1,F,сотрудник,0,102816.346412,профильное образование
705,-1,-902.084528,50,среднее,1,женат / замужем,0,F,госслужащий,0,137882.899271,приобретение автомобиля
742,-1,-3174.456205,57,среднее,1,женат / замужем,0,F,сотрудник,0,64268.044444,дополнительное образование
800,-1,349987.852217,54,среднее,1,Не женат / не замужем,4,F,пенсионер,0,86293.724153,дополнительное образование
941,-1,-1203.369529,57,Среднее,1,женат / замужем,0,F,пенсионер,0,145017.937533,на покупку своего автомобиля
1363,-1,-1195.264956,55,СРЕДНЕЕ,1,женат / замужем,0,F,компаньон,0,69550.699692,профильное образование
1929,-1,-1461.303336,38,среднее,1,Не женат / не замужем,4,M,сотрудник,0,109121.569013,покупка жилья
2073,-1,-2539.761232,42,среднее,1,в разводе,3,F,компаньон,0,162638.609373,покупка жилья
3814,-1,-3045.290443,26,Среднее,1,гражданский брак,1,F,госслужащий,0,131892.785435,на проведение свадьбы
4201,-1,-901.101738,41,среднее,1,женат / замужем,0,F,госслужащий,0,226375.766751,операции со своей недвижимостью


In [11]:
df[df['children'] > 5]

,children,days_employed,dob_years,education,education_id,family_status,family_status_id,gender,income_type,debt,total_income,purpose
606,20,-880.221113,21,среднее,1,женат / замужем,0,M,компаньон,0,145334.865002,покупка жилья
720,20,-855.595512,44,среднее,1,женат / замужем,0,F,компаньон,0,112998.738649,покупка недвижимости
1074,20,-3310.411598,56,среднее,1,женат / замужем,0,F,сотрудник,1,229518.537004,получение образования
2510,20,-2714.161249,59,высшее,0,вдовец / вдова,2,F,сотрудник,0,264474.835577,операции с коммерческой недвижимостью
2941,20,-2161.591519,0,среднее,1,женат / замужем,0,F,сотрудник,0,199739.941398,на покупку автомобиля
...,...,...,...,...,...,...,...,...,...,...,...,...
21008,20,-1240.257910,40,среднее,1,женат / замужем,0,F,сотрудник,1,133524.010303,свой автомобиль
21325,20,-601.174883,37,среднее,1,женат / замужем,0,F,компаньон,0,102986.065978,профильное образование
21390,20,-1203.369529,53,среднее,1,женат / замужем,0,M,компаньон,0,145017.937533,покупка жилой недвижимости
21404,20,-494.788448,52,среднее,1,женат / замужем,0,M,компаньон,0,156629.683642,операции со своей недвижимостью


In [12]:
mean_children = df['children'].mean()
df['children'].replace([-1, 20], mean_children, inplace=True)

В столбце `days_employed` имеются отрицательные значения, и можно их заменить абсолютным значением так как значения кажутся нормальными кроме знака минуса. Также имеются значения которые больше 21900 (365 дней * 60 лет), скорее всего это невозможно работать больше 60 лет. Есть два варианта:
* удалить строки которые больше 21900 
* поделить значения которые больше 21900 на 100

Я выберу деление так как это выглядит логично, может неправильно ввелись данные.

In [13]:
df[df['days_employed'] < 0]

,children,days_employed,dob_years,education,education_id,family_status,family_status_id,gender,income_type,debt,total_income,purpose
0,1.0,-8437.673028,42,высшее,0,женат / замужем,0,F,сотрудник,0,253875.639453,покупка жилья
1,1.0,-4024.803754,36,среднее,1,женат / замужем,0,F,сотрудник,0,112080.014102,приобретение автомобиля
2,0.0,-5623.422610,33,Среднее,1,женат / замужем,0,M,сотрудник,0,145885.952297,покупка жилья
3,3.0,-4124.747207,32,среднее,1,женат / замужем,0,M,сотрудник,0,267628.550329,дополнительное образование
5,0.0,-926.185831,27,высшее,0,гражданский брак,1,M,компаньон,0,255763.565419,покупка жилья
...,...,...,...,...,...,...,...,...,...,...,...,...
21519,1.0,-2351.431934,37,ученая степень,4,в разводе,3,M,сотрудник,0,115949.039788,покупка коммерческой недвижимости
21520,1.0,-4529.316663,43,среднее,1,гражданский брак,1,F,компаньон,0,224791.862382,операции с жильем
21522,1.0,-2113.346888,38,среднее,1,гражданский брак,1,M,сотрудник,1,89672.561153,недвижимость
21523,3.0,-3112.481705,38,среднее,1,женат / замужем,0,M,сотрудник,1,244093.050500,на покупку своего автомобиля


In [14]:
df['days_employed'] = df['days_employed'].abs()

In [15]:
df[df['days_employed'] > 21900]

,children,days_employed,dob_years,education,education_id,family_status,family_status_id,gender,income_type,debt,total_income,purpose
4,0.0,340266.072047,53,среднее,1,гражданский брак,1,F,пенсионер,0,158616.077870,сыграть свадьбу
18,0.0,400281.136913,53,среднее,1,вдовец / вдова,2,F,пенсионер,0,56823.777243,на покупку подержанного автомобиля
24,1.0,338551.952911,57,среднее,1,Не женат / не замужем,4,F,пенсионер,0,290547.235997,операции с коммерческой недвижимостью
25,0.0,363548.489348,67,среднее,1,женат / замужем,0,M,пенсионер,0,55112.757732,покупка недвижимости
30,1.0,335581.668515,62,среднее,1,женат / замужем,0,F,пенсионер,0,171456.067993,операции с коммерческой недвижимостью
...,...,...,...,...,...,...,...,...,...,...,...,...
21505,0.0,338904.866406,53,среднее,1,гражданский брак,1,M,пенсионер,0,75439.993167,сыграть свадьбу
21508,0.0,386497.714078,62,среднее,1,женат / замужем,0,M,пенсионер,0,72638.590915,недвижимость
21509,0.0,362161.054124,59,высшее,0,женат / замужем,0,M,пенсионер,0,73029.059379,операции с недвижимостью
21518,0.0,373995.710838,59,СРЕДНЕЕ,1,женат / замужем,0,F,пенсионер,0,153864.650328,сделка с автомобилем


In [16]:
df['days_employed'] = df['days_employed'].apply(lambda x: x/100 if x > 21900 else x)

В столбце `education` значения записей заглавными и не заглавными буквами. Нужно сделать все значения не заглавными.

In [17]:
df['education'].unique()

array(['высшее', 'среднее', 'Среднее', 'СРЕДНЕЕ', 'ВЫСШЕЕ',
       'неоконченное высшее', 'начальное', 'Высшее',
       'НЕОКОНЧЕННОЕ ВЫСШЕЕ', 'Неоконченное высшее', 'НАЧАЛЬНОЕ',
       'Начальное', 'Ученая степень', 'УЧЕНАЯ СТЕПЕНЬ', 'ученая степень'],
      dtype=object)

In [18]:
df['education'] = df['education'].str.lower()

В столбце `family_status` значения записей заглавными и не заглавными буквами. Нужно сделать все значения не заглавными.

In [19]:
df['family_status'].unique()

array(['женат / замужем', 'гражданский брак', 'вдовец / вдова',
       'в разводе', 'Не женат / не замужем'], dtype=object)

In [20]:
df['family_status'] = df['family_status'].str.lower()

В столбце `gender` имеется только одна строка со значением *'XNA'*, можно эту строку удалить.

In [21]:
df.groupby('gender').count()

,children,days_employed,dob_years,education,education_id,family_status,family_status_id,income_type,debt,total_income,purpose
gender,,,,,,,,,,,
F,14236,14236,14236,14236,14236,14236,14236,14236,14236,14236,14236
M,7288,7288,7288,7288,7288,7288,7288,7288,7288,7288,7288
XNA,1,1,1,1,1,1,1,1,1,1,1


In [22]:
df = df[df.gender != 'XNA']

В столбцах `education_id`, `family_status_id`, `income_type`, `debt`, `total_income`, `purpose` выглядит все нормально.

### Изменение типов данных

In [23]:
# Заменяем тип данных на целочисленный в столбцах где тип данных вещественный
for column in df.columns:
    if df[column].dtype == 'float64':
        df[column] = df[column].astype(int)

### Удаление дубликатов.

In [24]:
# Проверяем данные на дубликаты
df[df.duplicated()]

,children,days_employed,dob_years,education,education_id,family_status,family_status_id,gender,income_type,debt,total_income,purpose
2849,0,1203,41,среднее,1,женат / замужем,0,F,сотрудник,0,145017,покупка жилья для семьи
3290,0,1203,58,среднее,1,гражданский брак,1,F,пенсионер,0,145017,сыграть свадьбу
4182,1,1203,34,высшее,0,гражданский брак,1,F,сотрудник,0,145017,свадьба
4851,0,1203,60,среднее,1,гражданский брак,1,F,пенсионер,0,145017,свадьба
5557,0,1203,58,среднее,1,гражданский брак,1,F,пенсионер,0,145017,сыграть свадьбу
...,...,...,...,...,...,...,...,...,...,...,...,...
20702,0,1203,64,среднее,1,женат / замужем,0,F,пенсионер,0,145017,дополнительное образование
21032,0,1203,60,среднее,1,женат / замужем,0,F,пенсионер,0,145017,заняться образованием
21132,0,1203,47,среднее,1,женат / замужем,0,F,сотрудник,0,145017,ремонт жилью
21281,1,1203,30,высшее,0,женат / замужем,0,F,сотрудник,0,145017,покупка коммерческой недвижимости


Мы видим что в данных есть 71 строк дубликатов, следовательно мы удаляем эти дубликаты. Также видим что для этих строк `days_employed` и `total_income` столбцы были пусты так как они заполнены средним значением.

In [25]:
# Избавиляемся от дубликатов в данных
df.drop_duplicates(inplace=True)

### Формирование дополнительных датафреймов словарей, декомпозиция исходного датафрейма.

Создаём два новых датафрейма со столбцами:

- education_id и education — в первом;
- family_status_id и family_status — во втором.
    
Удаляем из исходного датафрейма столбцы education и family_status, оставив только их идентификаторы: education_id и family_status_id. Новые датафреймы — это те самые «словари», к которым мы сможем обращаться по идентификатору.

In [26]:
# Создаем датафрейм для education_id и education
df_education = pd.DataFrame({'education_id':[], 'education': []})
df_education['education_id'] = df['education_id']
df_education['education'] = df['education']

In [27]:
# Создаем датафрейм для family_status_id и family_status
df_family_status = pd.DataFrame({'family_status_id':[], 'family_status': []})
df_family_status['family_status_id'] = df['family_status_id']
df_family_status['family_status'] = df['family_status']

In [28]:
# Удаление из исходного датафрейма столбцы education и family_status
df.drop(['education', 'family_status'], axis=1, inplace=True)

### Категоризация дохода.

Создаём функию которая будет присваивать определенную категорию в столбец `total_income_category` относительно дохода клиента, то есть столбца `total_income`.

На основании диапазонов, указанных ниже, создаём столбец `total_income_category` с категориями:
   - 0–30000 — 'E';
   - 30001–50000 — 'D';
   - 50001–200000 — 'C';
   - 200001–1000000 — 'B';
   - 1000001 и выше — 'A'.
   
Например, кредитополучателю с доходом 25000 нужно назначаем категорию 'E', а клиенту, получающему 235000, — 'B'.

In [29]:
def income_category(income):
    if income <= 30000:
        return 'E'
    elif 30001 <= income <= 50000:
        return 'D'
    elif 50001 <= income <= 200000:
        return 'C'
    elif 200001 <= income <= 1000000:
        return 'B'
    elif income >= 1000001:
        return 'A'
    
df['total_income_category'] = df['total_income'].apply(income_category)

### Категоризация целей кредита.

Создаём функцию, которая на основании данных из столбца purpose сформирует новый столбец purpose_category, в который войдут следующие категории:
   - 'операции с автомобилем',
   - 'операции с недвижимостью',
   - 'проведение свадьбы',
   - 'получение образования'.
    
Например, если в столбце purpose находится подстрока 'на покупку автомобиля', то в столбце purpose_category появится строка 'операции с автомобилем'.

In [30]:
def purpose_category(purpose):
    if 'автомобил' in purpose:
        return 'операции с автомобилем'
    elif 'недвижимост' in purpose or 'жиль' in purpose:
        return 'операции с недвижимостью'
    elif 'свадьб' in purpose:
        return 'проведение свадьбы'
    elif 'образован' in purpose:
        return 'получение образования'
    
df['purpose_category'] = df['purpose'].apply(purpose_category)

# Ответьте на вопросы

Ответы на вопросы можно разместить в ячейках тетрадок Jupyter Notebook с типом markdown.

- Есть ли зависимость между количеством детей и возвратом кредита в срок?
- Есть ли зависимость между семейным положением и возвратом кредита в срок?
- Есть ли зависимость между уровнем дохода и возвратом кредита в срок?
- Как разные цели кредита влияют на его возврат в срок?
   
Ответы сопроводите интерпретацией — поясните, о чём именно говорит полученный вами результат.

##### Вопрос 1: Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [31]:
df.groupby('children')['debt'].agg(['count', 'sum', 'mean']).sort_values('mean')

,count,sum,mean
children,,,
5,9,0,0.000000
0,14213,1072,0.075424
3,330,27,0.081818
1,4808,444,0.092346
2,2052,194,0.094542
4,41,4,0.097561


##### Вывод 1: 
* Мы видем что у кого детей нет у них в процентном соотношений меньше не оплативших. 
* У кого пять детей их можно не считать так как очень мало данных всего лишь 9 строк.
* Выглядит проблематичным у кого 2 детей с процентном соотношением в 9.4 %

##### Вопрос 2: Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [32]:
df.groupby('family_status_id')['debt'].agg(['count', 'sum', 'mean']).sort_values('mean')

,count,sum,mean
family_status_id,,,
2,959,63,0.065693
3,1195,85,0.071130
0,12339,931,0.075452
1,4150,388,0.093494
4,2810,274,0.097509


In [33]:
df.pivot_table(index='family_status_id', values='debt', aggfunc='mean').sort_values('debt')

,debt
family_status_id,
2,0.065693
3,0.071130
0,0.075452
1,0.093494
4,0.097509


In [34]:
df_family_status['family_status'].unique()

array(['женат / замужем', 'гражданский брак', 'вдовец / вдова',
       'в разводе', 'не женат / не замужем'], dtype=object)

##### Вывод 2:
* Плохой результат по погашению кредита у `не женат / не замужем` почти 10%
* Самый хороший результат у `вдовец / вдова` почти 6.57%

##### Вопрос 3: Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [35]:
df.groupby('total_income_category')['debt'].agg(['count', 'sum', 'mean']).sort_values('mean')

,count,sum,mean
total_income_category,,,
D,350,21,0.060000
B,5040,356,0.070635
A,25,2,0.080000
C,16016,1360,0.084915
E,22,2,0.090909


In [36]:
df.pivot_table(index='total_income_category', values='debt', aggfunc='mean').sort_values('debt')

,debt
total_income_category,
D,0.060000
B,0.070635
A,0.080000
C,0.084915
E,0.090909


Cтолбец `total_income_category` с категориями:
   - 0–30000 — 'E';
   - 30001–50000 — 'D';
   - 50001–200000 — 'C';
   - 200001–1000000 — 'B';
   - 1000001 и выше — 'A'.

##### Вывод 3:
* У кого заработок между `30001-50000` имеют хорошую статистику по погашению кредита.
* Набольшие проблемные клиенты в категории А с заработком `1000001 и выше`

##### Вопрос 4: Как разные цели кредита влияют на его возврат в срок?

In [38]:
df.groupby('purpose_category')['debt'].agg(['count', 'sum', 'mean']).sort_values('mean')

,count,sum,mean
purpose_category,,,
операции с недвижимостью,10810,782,0.072340
проведение свадьбы,2324,186,0.080034
получение образования,4013,370,0.092200
операции с автомобилем,4306,403,0.093590


In [39]:
df.pivot_table(index='purpose_category', values='debt', aggfunc='mean').sort_values('debt')

,debt
purpose_category,
операции с недвижимостью,0.072340
проведение свадьбы,0.080034
получение образования,0.092200
операции с автомобилем,0.093590


##### Вывод 4:
* Погашение по кредиту по операциям с недвижимостью является лучшим с 7.23%
* Наихудчий результат по операциям с автомобилем с 9.36%

## Общий вывод:

Мы не можем уверенно сказать что определенный параметр сильно влияет на возврат кредита в срок. Так как нет сушественной разницы между категориями, максимум 1-3 процента. Также распределение данных разные по категориям. Например, очень мало людей которые зарабатывют больше 1 миллиона, и очень много людей с средним заработком. Такое распределение можно увидеть также для других категорий. 
- Относительно проведенной аналитики клиент с наибольшей вероятностью просрочки будет у кого `2 детей`, кто `не женат / не   замужем`, `с заработком 1000001 и выше`, и у того кто берёт кредит `по операциям с автомобилем`.

- Относительно проведенной аналитики клиент с хорошей истории по погашению кредита будет у кого `у кого детей нет`, кто `вдовец / вдова`, у кого заработок `между 30001-50000`, и у того кто берёт кредит `по операциям с недвижимостью`.